action.yml

```diff
@@ -0,0 +1,42 @@
+name: "Lagon CLI"
+description: "Easily integrate Lagon CLI operations into your Github workflow"
+branding:
+  icon: code
+  color: blue
+inputs:
+  lagon_token:
+    description: "Your Lagon API token"
+    required: true
+  command:
+    description: "The Lagon CLI command to run"
+    required: false
+    default: "deploy --prod"
+  site_url:
+    description: "Specify custom url for self-hosted Lagon instances"
+    required: false
+    default: "https://dash.lagon.app"
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Node.js"
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18
+        cache: npm
+
+    - name: "Install CLI"
+      run: npm install --global @lagon/cli esbuild
+
+    - name: "Execute CLI"
+      env:
+        LAGON_TOKEN: ${{ inputs.lagon_token }}
+        LAGON_COMMAND: ${{ inputs.command }}
+        LAGON_URL: ${{ inputs.site_url }}
+      shell: bash
+      run: |
+        # Setup config folder
+        mkdir -p ~/.lagon
+        # Setup auth
+        echo \{\"token\":\""${LAGON_TOKEN}"\",\"site_url\":\""${LAGON_URL}"\"\} >~/.lagon/config.json
```

Comment:
I've been thinking more about this: users that don't want to commit the `.lagon` folder won't be able to use this GitHub Action, because the config won't exist. I feel like it would be interesting to bring back inputs for `function_id`, `deployment_id`, `index` and `assets`. They could be optional but if provided, would create/override the config

Reply:
since we aren't requiring a `.lagon` folder in the action we can merge this then add support for no `.lagon` to make it more flexible.

Reply:
on second thought, I will add support for this back. Not committing `.lagon` makes a lot more sense tbh since you don't care about where it's deploy (function id) you just want to track what it does in git.